custom_components/airthings_wave/airthings.py

```diff
@@ -142,117 +141,112 @@ def decode_data(self, raw_data):
 
 class AirthingsWaveDetect:
     def __init__(self, scan_interval, mac=None):
-        self.adapter = pygatt.backends.GATTToolBackend()
         self.airthing_devices = [] if mac is None else [mac]
         self.sensors = []
         self.sensordata = {}
         self.scan_interval = scan_interval
         self.last_scan = -1
 
 
-    def find_devices(self):
-        # Scan for devices and try to figure out if it is an airthings device.
-        self.adapter.start(reset_on_start=False)
-        devices = self.adapter.scan(timeout=3)
-        self.adapter.stop()
-
-        for device in devices:
-            mac = device['address']
-            _LOGGER.debug("connecting to {}".format(mac))
-            try:
-                self.adapter.start(reset_on_start=False)
-                dev = self.adapter.connect(mac, 3)
-                _LOGGER.debug("Connected")
-                try:
-                    data = dev.char_read(manufacturer_characteristics.uuid)
-                    manufacturer_name = data.decode(manufacturer_characteristics.format)
-                    if "airthings" in manufacturer_name.lower():
-                        self.airthing_devices.append(mac)
-                except (BLEError, NotConnectedError, NotificationTimeout):
-                    _LOGGER.debug("connection to {} failed".format(mac))
-                finally:
-                    dev.disconnect()
-            except (BLEError, NotConnectedError, NotificationTimeout):
-                _LOGGER.debug("Faild to connect")
-            finally:
-                self.adapter.stop()
+    def _parse_serial_number(self, manufacturer_data):
+        try:
+            #print(manufacturer_data)
+            (ID, SN, _) = struct.unpack("<HLH", manufacturer_data)
+        except Exception as e:  # Return None for non-Airthings devices
+            return None
+        else:  # Executes only if try-block succeeds
+            if ID == 0x0334:
+                return SN
+
+    def find_devices(self, scanns=10):
+        scanner = btle.Scanner()
+        for _count in range(scanns):
+            advertisements = scanner.scan(0.1)
```

Comment:
Might change this defalut timeout 0.1s to configurable (should anyways not be a magic constant), maybe instead of the scann parameter? maybe good to have both configurable from Home Asistant config?